client/shaders/nodes_shader/opengl_fragment.glsl

```diff
@@ -22,11 +37,288 @@ varying mediump vec2 varTexCoord;
+				int end_offset = int(PCFSAMPLES) + init_offset;
+				
+				for ( int x=init_offset; x<end_offset; x++)
+				{
+					clampedpos = poissonDisk[x]*texture_size + smTexCoord.xy;
+					visibility += getHardShadowColor(shadowsampler, clampedpos.xy, realDistance);
+				}
+				
+				return visibility/PCFSAMPLES;
+			
+			}
+		#else
+			float getShadow(sampler2D shadowsampler, vec2 smTexCoord, float realDistance)
+			{
+				vec2 clampedpos;
+				float visibility=0.0;
+
+				float texture_size= 1/(f_textureresolution*0.25);
+				#if SHADOW_FILTER == 2
+					#define PCFBOUND 3.5
+					#define PCFSAMPLES 64.0
+				#elif  SHADOW_FILTER == 1
+					#define PCFBOUND 1.5
+					#define PCFSAMPLES 32.0
+				#else
+					#define PCFBOUND 0.0
+					#define PCFSAMPLES 4.0
+				#endif
+				int init_offset = int(floor(mod(((smTexCoord.x * 34.0) + 1.0) * smTexCoord.y, 64.0-PCFSAMPLES)));
+				int end_offset = int(PCFSAMPLES) + init_offset;
+				
+				for ( int x=init_offset; x<end_offset; x++)
+				{
+					clampedpos = poissonDisk[x]*texture_size + smTexCoord.xy;
+					visibility += getHardShadow(shadowsampler, clampedpos.xy, realDistance);
+				}
+
+				return visibility/PCFSAMPLES;
+			}
+		#endif
+
+	#else
+		#ifdef COLORED_SHADOWS
+			vec4 getShadowColor(sampler2D shadowsampler, vec2 smTexCoord, float realDistance)
+			{
+			
+				vec2 clampedpos;
+				vec4 visibility=vec4(0.0);
+
+				float texture_size= 1/(f_textureresolution*0.25);
+				#if SHADOW_FILTER == 2
+					#define PCFBOUND 3.5
+					#define PCFSAMPLES 64.0
+				#elif  SHADOW_FILTER == 1
+					#define PCFBOUND 1.5
+					#define PCFSAMPLES 16.0
+				#else
+					#define PCFBOUND 0.0
+					#define PCFSAMPLES 1.0
+				#endif
+				float y;
+				float x;
+				//basic PCF filter. we can explorer Poisson filter
+				for (y = -PCFBOUND ; y <=PCFBOUND ; y+=1.0)
+				for (x = -PCFBOUND ; x <=PCFBOUND ; x+=1.0)
+				{
+					clampedpos = vec2(x,y)*texture_size + smTexCoord.xy;
+					visibility+=getHardShadowColor(shadowsampler, clampedpos.xy, realDistance);
+				}
+				
+				return visibility/PCFSAMPLES;
+			}
+		#else
+			float getShadow(sampler2D shadowsampler, vec2 smTexCoord, float realDistance)
+			{
+				vec2 clampedpos;
+				float visibility=0.0;
+
+				float texture_size= 1/(f_textureresolution*0.25);
+				#if SHADOW_FILTER == 2
+					#define PCFBOUND 3.5
+					#define PCFSAMPLES 64.0
+				#elif  SHADOW_FILTER == 1
+					#define PCFBOUND 1.5
+					#define PCFSAMPLES 16.0
+				#else
+					#define PCFBOUND 0.0
+					#define PCFSAMPLES 1.0
+				#endif
+				float y;
+				float x;
+				//basic PCF filter. we can explorer Poisson filter
+				for (y = -PCFBOUND ; y <=PCFBOUND ; y+=1.0)
+				for (x = -PCFBOUND ; x <=PCFBOUND ; x+=1.0)
+				{
+					clampedpos = vec2(x,y)*texture_size + smTexCoord.xy;
+					visibility += getHardShadow(shadowsampler, clampedpos.xy, realDistance);
+				}
+				
+				return visibility/PCFSAMPLES;
+			}	
+		#endif
+	#endif
+#endif
+ 
+#ifdef ENABLE_TONE_MAPPING
```

Comment:
This is incorrect, `ENABLE_TONE_MAPPING` is *defined* but set to zero when disabled.
(Don't bother fixing this, I'll push a commit with it.)

Reply:
tnx :)